distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -151,5 +183,122 @@ private Response respond(Response response) {
 
             return response;
         }
+
+        /**
+         * Holds byte offset data specified by the request's range header and any related parsing
+         * methods.
+         */
+        private class ByteRange {
+            public int start;
+
+            public int end;
+
+            public ByteRange(int start, int end) {
+                this.start = start;
+                this.end = end;
+            }
+
+            /**
+             * Tokenizes the starting and ending bytes values from a range header. The stub server only
+             * advertising supporting byte offsets specifically, so this function assumes that byte
+             * ranges are provided and does not evaluate the unit of measurement
+             *
+             * @param rangeHeaderValue          Value of the range header sent in the request
+             * @param totalSizeOfProductInBytes Total size of the message to be returned. This is
+             *                                  not the partial content size, but the FULL size of
+             *                                  the product.
+             */
+            public ByteRange(String rangeHeaderValue, int totalSizeOfProductInBytes) {
+                // extract bytes
+                String startToken = StringUtils.substringBetween(rangeHeaderValue, "=", "-");
+                String endToken = StringUtils.substringAfter(rangeHeaderValue, "-");
+
+                // range offsets can be blank to indicate "until beginning" or "until end" of data.
+                if (StringUtils.isBlank(startToken)) {
+                    start = 0;
+                } else {
+                    start = Integer.parseInt(startToken);
```

Comment:
Should we catch the `NumberFormatException` here, log an exception and re-throw in case the `startToken` happens to _not_ be a number?